test/IECoreImage/ImageThinnerTest.py

```diff
@@ -46,14 +46,18 @@ def test( self ) :
 
 		IECoreImage.ImageThinner()( input=i, copyInput=False )
 
-		IECoreImage.ImageWriter( i, os.path.join( tempfile.gettempdir(), "newThinning.tif" ) ).write()
+		f = os.path.join( tempfile.gettempdir(), "newThinning.tif" )
 
-		ii = IECore.Reader.create( os.path.join( "test", "IECoreImage", "data", "tiff", "toTraceThinned.tif" ) ).read()
+		IECoreImage.ImageWriter( i, f ).write()
+
+		ii = IECore.Reader.create( f ).read()
+
+		iii = IECore.Reader.create( os.path.join( "test", "IECoreImage", "data", "tiff", "toTraceThinned.tif" ) ).read()
 
-		i.blindData().clear()
 		ii.blindData().clear()
+		iii.blindData().clear()
 
-		self.assertTrue( i==ii )
+		self.assertEqual( ii, iii )
```

Comment:
So, before this juggling, the test was failing? Doesn't that imply that something has changed in the way `ImageReader` is reading the files? Could this be related to the GL test problems? I'd like to understand what's going on here better...

Reply:
I can look into it more, but I can't guarantee I can fix it... note all other IECoreImage tests are passing still, so I'm not too concerned about OIIO being broken... your guess about the mysterious precision issues seem likely.

Reply:
Quick update in case I don't get further on this today, the problem is introduced here:
https://github.com/ImageEngine/cortex/blob/main/src/IECoreImage/ColorAlgo.cpp#L84

Reply:
I guess now we know that its an OCIO 2.0 sRGB -> linear conversion issue, the questions are (a) can this be reproduced outside of IE? and (b) how much do we care that 1.0 becomes 1.0000001192092896?

With my limited understanding of OCIO (and color conversion in general), I think we could not have been asserting exact image matches with anything other than 0 or 1 channel data previously, is that correct? If so, the precision limits in the GL tests seem fine and I could do the same in the ImageThinner if we don't like the extra write/read.

Reply:
> (a) can this be reproduced outside of IE?

I'll take a shot at this. My OCIO2 dependencies build just failed for some unknown reason, so I'm kicking the whole thing off from scratch. Will try to take a look later today.

> (b) how much do we care that 1.0 becomes 1.0000001192092896?

I think we should minimally : determine if it is OCIO or the OCIO config that has done this, and if it's OCIO, query it with them.

Reply:
> can this be reproduced outside of IE?

It can. I've reproduced here with OIIO `2.3.11.0` and OCIO `2.1.1`. It doesn't reproduce if I don't set the `OCIO` environment variable though, because then we hit the fall-back color transform from OIIO. So for what it's worth, this won't become an issue in our current CI setup.

For the uses of ImageReader I care about (loading icons in Gaffer, unit tests here), I would vote quite happily to avoid the OCIO config completely and use nothing but the basic lin/log/srgb transforms provided internally by OIIO (see https://github.com/ImageEngine/cortex/blob/main/src/IECoreImage/ImageReader.cpp#L469-L480). But I guess I can live with the relaxed checks if we document thoroughly in the commit message.

I also think it's worth querying the change in behaviour with the OCIO folks though - I'm no colour scientist but it doesn't seem right.

Reply:
The problem with running without OCIO is we skip several tests that could impact production, eg:
`testDPX (ImageWriterTest.ImageWriterTest) ... skipped 'Insufficient color specification. Linear -> Cineon conversion is not possible with an OCIO config'`

so I'm inclined to keep the relaxed checks. I'll squash the GL & Thinner commit together since they're about the same issue

Reply:
Addressed in 95927ac94eb574c949a07901cd27bfd4381691a9

Reply:
I've made a simple OCIO-only repro and reported it here : https://github.com/AcademySoftwareFoundation/OpenColorIO/issues/1600